libafl/src/events/launcher.rs

```diff
@@ -362,3 +365,246 @@ where
+    ) -> Result<(), Error>,
+    MT: Monitor + Clone,
+    SP: ShMemProvider + 'static,
+    S: DeserializeOwned + UsesInput,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Launcher")
+            .field("configuration", &self.configuration)
+            .field("broker_port", &self.broker_port)
+            .field("core", &self.cores)
+            .field("spawn_broker", &self.spawn_broker)
+            .field("remote_broker_addr", &self.remote_broker_addr)
+            .field("stdout_file", &self.stdout_file)
+            .field("stderr_file", &self.stderr_file)
+            .finish_non_exhaustive()
+    }
+}
+
+#[cfg(all(unix, feature = "std", feature = "fork"))]
+impl<'a, CF, MT, S, SP> CentralizedLauncher<'a, CF, MT, S, SP>
+where
+    CF: FnOnce(
+        Option<S>,
+        CentralizedEventManager<LlmpRestartingEventManager<S, SP>, SP>,
+        CoreId,
+    ) -> Result<(), Error>,
+    MT: Monitor + Clone,
+    S: DeserializeOwned + UsesInput + HasExecutions + HasClientPerfMonitor,
+    SP: ShMemProvider + 'static,
+{
+    /// Launch the broker and the clients and fuzz
+    #[allow(clippy::similar_names)]
+    #[allow(clippy::too_many_lines)]
+    pub fn launch(&mut self) -> Result<(), Error> {
+        if self.cores.ids.is_empty() {
+            return Err(Error::illegal_argument(
+                "No cores to spawn on given, cannot launch anything.",
+            ));
+        }
+
+        if self.run_client.is_none() {
+            return Err(Error::illegal_argument(
+                "No client callback provided".to_string(),
+            ));
+        }
+
+        let core_ids = get_core_ids().unwrap();
+        let num_cores = core_ids.len();
+        let mut handles = vec![];
+
+        log::info!("spawning on cores: {:?}", self.cores);
+
+        let stdout_file = self
+            .stdout_file
+            .map(|filename| File::create(filename).unwrap());
+        let stderr_file = self
+            .stderr_file
+            .map(|filename| File::create(filename).unwrap());
+
+        let debug_output = std::env::var("LIBAFL_DEBUG_OUTPUT").is_ok();
+
+        // Spawn centralized broker
+        self.shmem_provider.pre_fork()?;
+        match unsafe { fork() }? {
+            ForkResult::Parent(child) => {
+                self.shmem_provider.post_fork(false)?;
+                handles.push(child.pid);
+                #[cfg(feature = "std")]
+                log::info!("centralized broker spawned");
+            }
+            ForkResult::Child => {
+                log::info!("{:?} PostFork", unsafe { libc::getpid() });
+                self.shmem_provider.post_fork(true)?;
+
+                let mut broker: CentralizedLlmpEventBroker<S::Input, SP> =
+                    CentralizedLlmpEventBroker::on_port(
+                        self.shmem_provider.clone(),
+                        self.centralized_broker_port,
+                    )?;
+                broker.broker_loop()?;
```

Comment:
@domenukk I just start a second broker on another process, check this if it is correct

Reply:
Should work